api/document/views.py

```diff
@@ -23,7 +23,10 @@ class TreeView(GenericAPIView):
     queryset = DocNode.objects.none()   # Used to determine permissions
 
     def get_object(self, prefetch_related=True):
-        policy = policy_or_404(self.kwargs['policy_id'])
+        policy = policy_or_404(
+            self.kwargs['policy_id'],
+            only_public=not self.request.user.is_authenticated,
```

Comment:
Overall I think this PR is very straightforward, but I wonder if it would be better to create a variable outside of this call so as to not have to follow the `not` logic in a single line?

Reply:
Good idea, I'll add. I'll try to resolve the cyclomatic complexity warning, too.